src/hooks/FilterData.jsx

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import { COUNT_ABOVE, COUNT_BELOW } from "../utils/data";
+
+const FilterData = () => {
+  const { isCountAboveOrBelow, endDate } = useSelector(
+    (store) => store.fakeapidata
+  );
+  const data = useSelector((store) => store.fakeapidata.data);
+  const startDate = useSelector((store) => store.fakeapidata.startDate);
+
+  const [filteredData, setFilteredData] = useState(data);
+
+  const filterData = () => {
+    let newFilteredData = data;
+
+    if (isCountAboveOrBelow) {
+      newFilteredData = data?.filter((item) => {
+        if (
+          (isCountAboveOrBelow === COUNT_BELOW && item?.rating?.count <= 100) ||
```

Comment:
@9xVibee you can return it directly

```
ata?.filter((item) => (isCountAboveOrBelow === COUNT_BELOW && item?.rating?.count <= 100) || (isCountAboveOrBelow === COUNT_ABOVE && item?.rating?.count > 100))
```

Reply:
@Febinvarghese2 Actually if we directly return without checking if filter is applied or not that case it will give me empty array!